service/routes.py

```diff
@@ -29,3 +29,35 @@ def index():
 ######################################################################
 
 # Place your REST API code here ...
+
+
+# -----------------------------------------------------------
+# Create counters
+# -----------------------------------------------------------
+@app.route("/shopcarts/<customer_id>/<item_id>", methods=["POST"])
+def add_item(customer_id, item_id):
+    """Creates a new entry and stores it in the database
+    Args:
+        customer_id (str): the id of the customer and item to add for it
+        item_id (str): the id of item to be added
+    Returns:
+        dict: the row entry in databse which contains customer_id, item_id and quantity default to 1
+    """
+    app.logger.info(f"Request to add item for customer {customer_id} and item {item_id}")
+    if customer_id is None or item_id is None or not customer_id.isdigit() or not item_id.isdigit() :
+        abort(status.HTTP_400_BAD_REQUEST, f"Bad request for {customer_id} {item_id}")
+
+    customer_id = int(customer_id)
+    item_id = int(item_id)
+    
+    if ShopCarts.check_exist_by_customer_id_and_product_id(customer_id, item_id):
```

Comment:
should we maybe have a unique key constraint on the table instead of checking in the code?

Reply:
Each shop record is unique base on the customer and item. It needs 2 tables to implement the unique constrain.

Reply:
i meant we can have a unique key constraint on the combination of customer id and product id in the same table. So the table would look like this with **customer_product_index** as the unique constraint and the code doesn't need to check for the condition.
```
   Column    |  Type   | Collation | Nullable |             Default              | Storage | Compression | Stats target | Description 
-------------+---------+-----------+----------+----------------------------------+---------+-------------+--------------+-------------
 id          | integer |           | not null | generated by default as identity | plain   |             |              | 
 customer_id | integer |           |          |                                  | plain   |             |              | 
 product_id  | integer |           |          |                                  | plain   |             |              | 
 quantities  | integer |           |          |                                  | plain   |             |              | 
Indexes:
    "shop_carts_pkey" PRIMARY KEY, btree (id)
    "ix_shop_carts_customer_id" btree (customer_id)
    "customer_product_index" UNIQUE CONSTRAINT, btree (customer_id, product_id)
```